docs/refman/generatorbase.py

```diff
@@ -2,14 +2,14 @@
 # Copyright 2021 The Meson development team
 
 
-from abc import ABCMeta, abstractmethod
+from abc import abstractmethod
 import typing as T
 
 from .model import ReferenceManual, Function, Method, Object, ObjectType, NamedObject
 
 _N = T.TypeVar('_N', bound=NamedObject)
 
-class GeneratorBase(metaclass=ABCMeta):
+class GeneratorBase:
```

Comment:
Just to learn something: is it valid to use the `@abstractmethod` decorator in a class that whose metaclass is not `ABCMeta`? The Python documentation says that it is required, but it does not say why it is https://docs.python.org/3/library/abc.html#abc.abstractmethod Quick testing shows that using `@abstractmethod` in a class whose metaclass is not `ABCMeta` has no effect when the class is instantiated: no exception about the missing implementation.